src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -254,15 +255,22 @@ def check_exists(changed_headers, parser, values, proj, missing_headers, i):
             # Check if the next file is in missing_headers
             return check_exists(changed_headers, parser, values, proj, missing_headers, i + 1)
         else:
+            # Save current copy of files[i]
+            tempfile = NamedTemporaryFile(mode="w", delete=True).name
+            shutil.copyfile(files[i], tempfile)
+
             # Update the header
-            with NamedTemporaryFile(mode="w", delete=False) as tmp:
+            with NamedTemporaryFile(mode="w", delete=True) as tmp:
                 args = set_header_args(parser, year, files[i], copyright, template)
                 header.run(args, proj, tmp)
                 tmp.close()
 
```

Comment:
I fail to understand what you are doing here. You are creating a ``tmp`` file which you are modifying and then deleting? You are effectively doing no changes, right?

Reply:
For some reason I thought it deleted the temporary files once the script was done running rather than once that section of code was done running. In that case, I can change them back to False